### PR TITLE
Feature tidy up

### PR DESCRIPTION
**Pull request by Heath101**
Might make sense to review these changes commit by commit, so that there is some context to the changes

Removed unused dependencies
Removed dependencies with security vulnerabilities
Updated a few dependencies
Replaced monologue with EventEmitter
Removed metrics instrumentation dev dependency
Fixed some eslint things
Converted to use native promises
Consolidated to use shared auto required test setup

Testing steps:

1. Pull down the project and run the tests
2. Open a project that uses seriate, and npm link seriate to a local version of this PR.
3.  Regression test that all the functionality works in the project the way that it should, and no errors are encounter relating to seriate.

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+8.15
```
Comment (veryfancy) [inline .nvmrc]: Ha, was going to suggest adding this as I got my bearings in seriate repo. Good call.

**Comment by veryfancy** [comment]
`npm depcheck` output:
```
Missing dependencies
* tedious
```
Think we should add it? I gather it'll come along with the `mssql` package. But it is referenced directly.

File: spec/behavior/external-sql.spec.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/***************************************************
+/** *************************************************
```
Comment (veryfancy) [inline spec/behavior/external-sql.spec.js]: This appears unintentional.

Comment (Heath101) [reply spec/behavior/external-sql.spec.js]: danged ES lint :)

**Pull request merged**